js/src/Ice/TcpTransceiver.js

```diff
@@ -69,20 +75,7 @@ if (typeof net.createConnection === "function") {
                     //
                     // Socket is connected.
                     //
-                    this._desc = fdToString(this._fd, this._proxy, this._addr);
-                    this._state = StateConnected;
-                } else if (this._state === StateProxyConnectRequest) {
```

Comment:
The `StateProxyConnectRequest` and `StateProxyConnectRequestPending` were never entered. And `this._proxy` was never assigned.

Seems this come from the initial port but was never implemented. Related to Socks and Http proxy support with Ice.